controlplane/src/core/repositories/analytics/TraceRepository.ts

```diff
@@ -78,4 +78,45 @@ export class TraceRepository {
       },
     }));
   }
+
+  public async getRouterVersion(organizationID: string, startDate: string): Promise<string> {
+    const parsedDateRange = isoDateRangeToTimestamps({ start: startDate, end: startDate });
+    const [start] = getDateRange(parsedDateRange);
+
+    const queryWithTimeConstraint = `
+    WITH 
+      toDateTime('${start}') AS startDate
```

Comment:
We have no end date?

Reply:
no its not required, as i only care about what the router version was when the composition happened

Reply:
Can we limit it somehow to an end date because it has drastic impact on the performance of the query. A composition usually has an end date as long as it is not the latest.